sdk/nodejs/appservice/zMixins.ts

```diff
@@ -501,3 +501,264 @@ export function getResourceGroupNameAndLocation(
     const getResult = resourceGroupName.apply(n => core.getResourceGroup({ name: n }));
     return { resourceGroupName, location: getResult.location };
 }
+
+export type CallbackArgs<C extends Context<R>, E, R extends Result> = {

```

Comment:
Can we instead extract this part out from the existing CallbackFunctionAppArgs?

Reply:
`CallbackArgs` is intended for an individual Azure Function, so all other parameters of `FunctionAppArgs` do not apply. I agree this deserves a clean-up, eventually.

Reply:
What i'm saying is:  Since you extracted this, teh normal CallbackFunctionAppArgs should then derive from this type, so we don't have the duplication.  Thanks! :)

Reply:
How do I derive `CallbackFunctionAppArgs` from FunctionAppArgs and CallbackArgs with util.Overwrite? Do I need an intermediate
```
interface FunctionAppArgsWithCallback<C extends Context<R>, E, R extends Result> 
    extends FunctionAppArgs, CallbackArgs<C, E, R> {
}
export type CallbackFunctionAppArgs<C extends Context<R>, E, R extends Result> = 
    util.Overwrite<FunctionAppArgsWithCallback<C, E, R>, {
...
```
?

Reply:
So, i'm thinkin like this (names not final):

```ts
interface BaseFunctionAppArgs<C, E, R> {
    callback?: Callback<C, E, R>;
    callbackFactory?: CallbackFactory<C, E, R>;
}

type CallbackFunctionAppArgs<C, E, R> = BaseFunctionAppArgs<C, E, R> & utils.Overwrite<...
```

Note: we should likely move away from utils.Overwrite for this package as well (we already did for awsx).  THe problem is that it makes the docs and type-sigs really hard for normal mortals to read :)   We isntead use interfaces.  This means we do need to copy over the values we care about from teh core interfaces, but it makes it super easy to grok.  In this case, it would then look like:

```ts
interface BaseFunctionAppArgs<C, E, R> {
    callback?: Callback<C, E, R>;
    callbackFactory?: CallbackFactory<C, E, R>;
}

interface CallbackFunctionAppArgs<C, E, R> extends BaseFunctionAppArgs<C, E, R>, SomeOtherCommonInterface {
     // add additional members here manually
}
```